ntbs-service/wwwroot/source/Components/YearComparison.ts

```diff
@@ -10,6 +10,10 @@ const YearComparison = Vue.extend({
             const inputField = event.target as HTMLInputElement;
             const newValue = inputField.value;
 
+            if (!newValue) {
+                return;
+            }

```

Comment:
By doing this, I think you've bypassed the code which clears the error message. So if you enter an invalid date and then delete it we'd still be displaying the error message?

I think it would be more consistent to fix this by still making the validation request and returning an OK response.

Reply:
What do you think?

Reply:
I think it would be cleaner to have the error cleared yes, but we have cases for both throughout the site (some fields do remove error messages, some do not). Not a huge issue but should we be trying to move to this method?

Reply:
I definitely think we should always be removing error messages when the state returns to something that is valid.

There's a question of whether we do that via that backend or the frontend. Doing it in the backend is probably simpler, but does create an unnecessary API request here.